### PR TITLE
php: add switch to disable update checking

### PR DESCRIPTION
**Pull request by felipe19930**
It's controlled via the `IGNORE_UPDATES` environment variable. Set it to `1` to disable update checking.

### CODE REVIEW

**Comment by xPaw** [comment]
Why?

**Comment by hectorevil** [comment]
autoupdates maybe?

**Comment by felipe19930** [comment]
@xPaw I'm running about 50 instances of the script and the update check routine runs at each 2 minutes, so my terminal window gets a bit polluted. I'd like a way to disable it since it can be a bit annoying sometimes. This is more of a feature for "power users", since the update check is still enabled by default...

File: cheat.php [hunk 1]
```diff
@@ -778,3 +789,4 @@ function Msg( $Message, $EOL = PHP_EOL, $printf = [] )
 		echo $Message;
 	}
 }
+# vim: set tabstop=8 softtabstop=8 shiftwidth=8 noexpandtab :
```
Comment (xPaw) [inline cheat.php]: undo this

Comment (felipe19930) [reply cheat.php]: Are you sure? It's just a comment to help VIM users like myself who don't use tabulations. When I hit TAB, the VIM editor converts them to 4 spaces instead. This line I added just prevents that...

Comment (xPaw) [reply cheat.php]: If we stored configs for each unique editor, it would be a mess.

Comment (felipe19930) [reply cheat.php]: Fair enough!

**Comment by PsyBooty** [comment]
@RageGirl on php add `$_SERVER[ 'IGNORE_UPDATES' ] = true;` before `if( isset( $_SERVER[ 'IGNORE_UPDATES' ] ) && (bool)$_SERVER[ 'IGNORE_UPDATES' ] )`

**Comment by felipe19930** [comment]
@PsyBooty's suggestion works, but you'd have add that change to `cheat.php` every time there's an update. You can do that differently: Open `cheat.bat` with a text editor ([Notepad++](https://notepad-plus-plus.org/) is recommended). At the end of the file, you will see this:

```bat
:start
php\php.exe -f cheat.php
goto start
```

So you replace that by the following:

```bat
:start
set IGNORE_UPDATES=1
php\php.exe -f cheat.php
goto start
```

Now every time you run `cheat.bat`, update checking will be disabled.

**Comment by PsyBooty** [comment]
@felipe19930 hmm nice...
Can u do it on the .htaccess on the folder like that??
<VirtualHost *:80>
    ...
    SetEnv IGNORE_UPDATES "1"
    ...
</VirtualHost>

**Comment by felipe19930** [comment]
I'm not sure. I didn't know people were running this script through a web server. How does that work by the way? I assume you would have to load the cheat.php page in your web browser and leave it open, otherwise the script will stop its execution?

**Comment by PsyBooty** [comment]
@felipe19930 u cant, just want to know if would work =)

**Comment by felipe19930** [comment]
Ah right. Maybe it could work :) I just don't see why do it though.

**Comment by RageGirl** [comment]
@felipe19930 and @PsyBooty  thanks alot for those guidelines. + Respect

**Comment by Botan626** [comment]
@felipe19930 How to add "set IGNORE_UPDATES=1"  to a bat like this:

cd c:\php\
start php SalienCheat\cheat.php token
start php SalienCheat\cheat.php token

?

**Pull request merged**